### PR TITLE
SOLR-16039: Upgrade to Hadoop 3.3.2

### PR DESCRIPTION
**Pull request by risdenk**
https://issues.apache.org/jira/browse/SOLR-16039

### CODE REVIEW

**Comment by HoustonPutman** [comment]
Thanks for getting this started. Will be nice if we can help them test the RC5, and give our input 🙂

**Comment by risdenk** [comment]
Errr looks like the rc4 maven repo went away in the meantime :D

File: versions.props [hunk 1]
```diff
@@ -25,7 +25,7 @@ io.opentracing:*=0.33.0
 io.prometheus:*=0.2.0
 javax.servlet:javax.servlet-api=3.1.0
 joda-time:joda-time=2.9.9
-junit:junit=4.13.1
+junit:junit=4.13.2
```
Comment (madrob) [inline versions.props:28]: I'm surprised that we need this here? I'm not against it, just found it odd that our unit version is driven by our Hadoop version.

Comment (risdenk) [reply versions.props:28]: So Hadoop bumped the version from 4.13.1 to 4.13.2 (based on gradle resolution rules) - I just made it explicit here.

File: solr/modules/hdfs/src/test/org/apache/hadoop/util/DiskChecker.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
 import java.util.UUID;
 import java.util.concurrent.atomic.AtomicReference;
 
-import com.google.common.annotations.VisibleForTesting;
+import org.apache.hadoop.thirdparty.com.google.common.annotations.VisibleForTesting;
```
Comment (madrob) [inline solr/modules/hdfs/src/test/org/apache/hadoop/util/DiskChecker.java:28]: Did Hadoop switch to shading google-annotations? I thought we already had our own copy of it anyway.

Comment (risdenk) [reply solr/modules/hdfs/src/test/org/apache/hadoop/util/DiskChecker.java:28]: Guava is shaded by Hadoop. I don't know that it matters where we get the Guava annotations from. We still pull in Guava for the HDFS block cache stuff, but I figured it made more sense to pull in the Hadoop shaded Gauva to keep the `org.apache.hadoop` files closer to the original.

**Comment by risdenk** [comment]
3.3.2 was released - https://lists.apache.org/thread/z521vnzqv7vdv4gtzszbt35lrbnrovty. I'll update this PR shortly

**Comment by risdenk** [comment]
This is ready for review.

**Review by madrob — APPROVED** [review]

**Pull request merged**